test/clients/retry_client.go

```diff
@@ -17,97 +17,97 @@ package clients
 
 import (
 	"context"
-	"time"
 
-	"k8s.io/apimachinery/pkg/runtime"
+	"golang.org/x/time/rate"
+	"k8s.io/apimachinery/pkg/api/meta"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
 )
 
 // RetryClient wraps a client to provide rate-limiter respecting retry behavior.
 type RetryClient struct {
+	// mu      sync.RWMutex
+	Limiter *rate.Limiter
 	client.Client
 }
 
 // retry will run the provided function, retrying if it fails due to rate limiting.
-// It will respect the rate limiters delay guidance. If context is cancelled, it will
-// return early.
-func retry(ctx context.Context, f func() error) error {
-	for {
-		if err := ctx.Err(); err != nil {
-			return err
-		}
-		err := f()
-		if delay, needDelay := apiutil.DelayIfRateLimited(err); needDelay {
-			select {
-			case <-ctx.Done():
-				return ctx.Err()
-			case <-time.After(delay):
-				continue
-			}
-		}
+// If context is cancelled, it will return early.
+func retry(ctx context.Context, limiter *rate.Limiter, f func() error) error {
+	if err := ctx.Err(); err != nil {
```

Comment:
The for loop was removed - I'm not sure this retries anything anymore.

Reply:
fixed